lib/ansible/plugins/callback/__init__.py

```diff
@@ -412,3 +412,10 @@ def v2_runner_item_on_skipped(self, result):
 
     def v2_runner_retry(self, result):
         pass
+
+    def v2_runner_on_start(self, host, task):
```

Comment:
All of the other `v2_runner_on` methods just pass `self, result` for arguments. When host is needed it does `host = result._host.get_name()`

Is there any reason to do this differently? I'm concerned about the stability of the call pattern when it's starkly different like this.

Reply:
There is no `result` before the task completes. This indicates the start of a host executing, so `host` and `task` are provided so that you have that data available for association.  Later when the task for the host completes, you get a `result` which effectively combines the `host`, the `task` and the `result` into 1 object.

Reply:
@AlanCoding since most of those deal with 'after' and this is meant to deal with 'before' the signature is not expected to be consistent.

Reply:
okay, that makes sense. Typically we utilize the host, task, and result from the result object, but seeing as how the 3rd does not exist in this case I see what you are doing.